weather_providers/weathergov.py

```diff
@@ -0,0 +1,97 @@
+                    "tornado": "tornado_hurricane",
+                    "hurricane": "tornado_hurricane",
+                    "tropical_storm": "tornado_hurricane",
+                    "dust": "dust_ash_sand",
+                    "smoke": "fire_smoke",
+                    "haze": "haze",
+                    "hot": "very_hot",
+                    "cold": "cold",
+                    "blizzard": "blizzard",
+                    "fog": "climacell_fog"
+                    }
+
+        return icon_dict[weathergov_icon]
+
+    def get_forecast_url(self, lat, long):
+        logging.info("Using lat long to figure out the Weather.gov forecast URL")
+        lookup_url = "https://api.weather.gov/points/{},{}".format(lat, long)
+        lookup_data = get_json_from_url(lookup_url, {'User-Agent':'({0})'.format(self.weathergov_self_id)}, "cache_weather_gov_lookup.json", 3600)
+        logging.debug(lookup_data)
+        return lookup_data["properties"]["forecast"]
+
+    # Get weather from Weather.Gov, US only
+    # https://www.weather.gov/documentation/services-web-api
+    def get_weather(self):
+
+        forecast_url = self.get_forecast_url(self.location_lat, self.location_long)
+        # https://api.weather.gov/gridpoints/TOP/31,80/forecast"
+        logging.info(forecast_url)
+
+        response_data = self.get_response_json(forecast_url, {'User-Agent':'({0})'.format(self.weathergov_self_id)})
+        weather_data = response_data
+        logging.debug("get_weather() - {}".format(weather_data))
+
+        daytime = self.is_daytime(self.location_lat, self.location_long)
+
+        # Weather.gov doesn't provide a min max temperature.  It uses the today and tonight temperature as min max instead.  
+        for forecast in weather_data["properties"]["periods"]:
+            if forecast["name"]=="Today":
+                day_forecast = forecast
+            if forecast["name"]=="Tonight":
+                night_forecast = forecast
+
+        # {'number': 2, 'name': 'Tonight', 'startTime': '2022-03-06T18:00:00-06:00', 'endTime': '2022-03-07T06:00:00-06:00', 'isDaytime': False, 'temperature': 20, 'temperatureUnit': 'F', 'temperatureTrend': None, 'windSpeed': '10 to 15 mph', 'windDirection': 'N', 'icon': 'https://api.weather.gov/icons/land/night/snow,30/snow,20?size=medium', 'shortForecast': 'Chance Rain And Snow', 'detailedForecast': 'A chance of rain and snow before 3am. Mostly cloudy, with a low around 20. North wind 10 to 15 mph, with gusts as high as 25 mph. Chance of precipitation is 30%.'}
+        current_forecast = day_forecast if daytime else night_forecast
+
+        weather = {}
+        weather["description"] = current_forecast["shortForecast"]
+        weather["temperatureMin"] = night_forecast["temperature"] if self.units != "metric" else self.f_to_c(night_forecast["temperature"])
+        weather["temperatureMax"] = day_forecast["temperature"] if self.units != "metric" else self.f_to_c(day_forecast["temperature"])
```

Comment:
The API doesn't return a min/max temperature. So I'm taking the 'today' temperature as high, and 'tonight' temperature as low.